vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,565 @@
+    filter(time_value < 2017),
+  outcome = "num_graduates",
+  predictors = c("num_graduates"),
+  args_list = arx_args_list(
+    lags = list(c(0, 1, 2)),
+    ahead = 1
+  )
+)
+# and plotting
+autoplot(
+  grad_forecast$epi_workflow,
+  grad_forecast$predictions,
+  grad_employ,
+)
+```
 
-```{r}
-extract_frosting(out_q$epi_workflow)
+The 8 graphs represent all combinations of the `geo_values` (`"Quebec"` and `"British Columbia"`), `edu_quals` (`"Undergraduate degree"` and `"Professional degree"`), and age brackets (`"15 to 34 years"` and `"35 to 64 years"`).
+
+## Fitting a non-geo-pooled model
+
+The methods shown so far fit a single model across all geographic regions.
+This is called "geo-pooling". 
+To fit a non-geo-pooled model that fits each geography separately, one either needs a multi-level
+engine (which at the moment `{parsnip}` doesn't support), or one needs to loop over
+geographies.
+Here, we're using `purrr::map` to perform the loop.
+
+```{r fit_non_geo_pooled, warning=FALSE}
+geo_values <- covid_case_death_rates |>
+  pull(geo_value) |>
+  unique()
+
+all_fits <-
+  purrr::map(geo_values, \(geo) {
+    covid_case_death_rates |>
+      filter(
+        geo_value == geo,
+        time_value <= forecast_date
+      ) |>
+      arx_forecaster(
+        outcome = "death_rate",
+        trainer = linear_reg(),
+        predictors = c("death_rate"),
+        args_list = arx_args_list(
+          lags = list(c(0, 7, 14)),
+          ahead = 14
+        )
+      )
+  })
+map_df(all_fits, ~ pluck(., "predictions"))
 ```
 
-Here we have 5 layers of frosting. The first generates the forecasts from the test data.
-The second uses quantiles of the residuals to create distributional
-forecasts. The next two add columns for the date the forecast was made and the
-date for which it is intended to occur. Because we are predicting rates, they
-should be non-negative, so the last layer thresholds both predicted values and
-intervals at 0. The code to do this (inside the forecaster) is
-
-```{r}
-f <- frosting() %>%
-  layer_predict() %>%
-  layer_residual_quantiles(
-    quantile_levels = c(.01, .025, 1:19 / 20, .975, .99),
-    symmetrize = TRUE
-  ) %>%
-  layer_add_forecast_date() %>%
-  layer_add_target_date() %>%
-  layer_threshold(starts_with(".pred"))
+Fitting separate models for each geography is both 56 times slower[^7] than geo-pooling, and fits each model on far less data.
+If a dataset contains relatively few observations for each geography, fitting a geo-pooled model is likely to produce better, more stable results.
+However, geo-pooling can only be used if values are comparable in meaning and scale across geographies or can be made comparable, for example by normalization.
+
+If we wanted to build a geo-aware model, such as a linear regression with a different intercept for each geography, we would need to build a [custom workflow](custom_epiworkflows) with geography as a factor.
+
+# Anatomy of a canned forecaster
```

Comment:
I'm against empty sections. Maybe add a sentence or two setting up the discussion?